### ISSUE TITLE
how to choose final genome output file of homozygous with hic

### ISSUE THREAD

**Issue by Lillian-21** [chhylp123/hifiasm#198]
I have a homozygous plant sepceis (estimatic size 1.1G). 
First, I run hifi read (33G) with hifiasm-0.16.1/hifiasm  -o out.fasta -t 16 -l0 input.fasta. I get 1.8G p_ctg (N50 2.3M, not very good), 158M a_ctg. Then I used p_ctg to run Hi-C phasing: hifiasm-0.16.1/hifiasm -l0 -o p_ctg.fa -t 24 --h read1.fq.gz -h2 read2.fq.gz. I get 1.3G hic.p_ctg.fa;  1.3G hap1.p_ctg.fa; 1.2G hap2.p_ctg.fa.
Now I do not know which one should I use to do next analysis? which one is the whole geneme? How to understand the hap1 and hap2?
Besides, I get a abnomal kmer.

<img width="860" alt="20211019102143" src="https://user-images.githubusercontent.com/71699451/137833340-0a1527bd-679a-47a0-a6e4-010c1ddc96e1.png">
<img width="783" alt="20211019102215" src="https://user-images.githubusercontent.com/71699451/137833344-3d8ed655-0a73-4d8e-b0bb-fb102e703b3f.png">
<img width="877" alt="20211019102235" src="https://user-images.githubusercontent.com/71699451/137833347-489762c2-246d-43c9-a627-b018038dc9d3.png">

**Comment by chhylp123** [comment]
Does your sample just has one haplotype with 8x coverage?

**Comment by B10inform** [comment]
Hi chhylp123,

How important is it and what are the difference phasing homozygous genome vs heterozygous genome?

**Comment by chhylp123** [comment]
I guess there is no need to phase homozygous genomes, as they only have one haplotype?

**Comment by B10inform** [comment]
Should the homozygous genomes be purged? Without purging the genome size would have lots of duplicates. 

Heterozygous: Genome size 600Mb, phasing gives hap1 ~ 300Mb hap2 ~ 300Mb, both are used for downstream analysis
Homozygous: Genome size 600Mb, _(if phase gives hap1 ~ 300Mb hap2 ~ 300Mb)_ Use 600Mb for downstream analysis??

Thanks

**Comment by chhylp123** [comment]
Just make sure: what are the differences between the homozygous genome and the heterozygous genome? If a genome is homozygous, hifiasm with `-l0` should not produce assembly including lots of duplicates.

**Comment by B10inform** [comment]
Property | min | max
-- | -- | --
Homozygous   (aa) | 99.09% | 99.11%
Heterozygous   (ab) | 0.88% | 0.90%

hifiasm -o sample.asm -l0 sample.fastq

.p_ctg.fasta + .a_ctg.fasta = .pa_ctg.fasta (600Mb)
Genome size 600Mb, 

if phase: hap1 ~ 300Mb hap2 ~ 300Mb.

What to use for downstream analysis?? 600Mb or ~300Mb?

Thanks

**Comment by chhylp123** [comment]
Could you please check the assembly graph of the homozygous genome? If the graph has a lot of small bubbles, it is more likely to be a heterozygous genome with low heterozygosity. In this case, I would recommend you to use phased assemblies.

**Comment by B10inform** [comment]
This is how the assembly graph looks like.

![image](https://user-images.githubusercontent.com/66449329/157980274-96b74909-1365-418e-bac1-32599501c517.png)

**Comment by chhylp123** [comment]
I mean the `p_utg.noseq.gfa`, which could be visualized by Bandage. But at least from your k-mer plot, it is a heterozygous genome.

**Comment by B10inform** [comment]
the p_utg.noseq.gfa

![image](https://user-images.githubusercontent.com/66449329/157982438-59c66616-b8f5-49e1-9b52-27b3ea653a70.png)

**Comment by chhylp123** [comment]
Well, what's the estimated genome size by k-mers, and the BUSCO scores? I guess it should be a heterozygous genome with high heterozygosity. But there is a slight possibility that this genome is homozygous.

**Comment by B10inform** [comment]
This is p_ctg.noseq.gfa
![image](https://user-images.githubusercontent.com/66449329/157983920-84d6f2bc-a1e8-4ebb-8ef0-021eda64537a.png)

**Comment by B10inform** [comment]
The estimated genome size is 301Mb.

With K-mer GEnomeeScope
![image](https://user-images.githubusercontent.com/66449329/157984161-a4390a01-5b21-43f8-84df-c1c3848bcc1c.png)

**Comment by chhylp123** [comment]
Then I guess it is a heterozygous genome.

**Comment by B10inform** [comment]
So i should not use -l0.

**Comment by chhylp123** [comment]
Yes, I guess phased assemblies should work. You could also compare the BUSCO scores for double checking.

**Comment by B10inform** [comment]
Some confusion:
**Heterozygous species:**

**GenomeScope:**
Property | min
-- | --
Homozygous   (aa) | 96.44%
Heterozygous   (ab) | 3.51%

![image](https://user-images.githubusercontent.com/66449329/157987057-e980273e-97c7-4426-9071-8d8c735d6030.png)

Assembly graph
![image](https://user-images.githubusercontent.com/66449329/157986982-4fce77d4-e9b0-4781-b2ac-27bcdc821121.png)

**Homozygous species:**

GenomeScope graph and table shows Homozygous:  

Property | min
-- | --
Homozygous   (aa) | 99.09%
Heterozygous   (ab) | 0.88%

![image](https://user-images.githubusercontent.com/66449329/157987164-ef3e05cb-21ff-4ab3-9279-4d6bf5bd852d.png)

Assembly graph
![image](https://user-images.githubusercontent.com/66449329/157987406-1b8c1c5d-eb1f-4b4f-98fc-b97593f42d59.png)


The look totally different.How to reliably confirm the heterozygosity and homozygocity?

Thanks

**Comment by chhylp123** [comment]
The reason is that GenomeScope and hifiasm utilize k-mers with different lengths. I guess most genomes are heterozygous unless some very special genomes. For these genomes, you should always know they are homozygous in advance.
